static/js/ElasticSearchQuerier.jsx

```diff
@@ -0,0 +1,376 @@
+    async addRefTopic(topic) {
+        const book = await Sefaria.getIndexDetails(topic.key);
+        return {
+            enDesc: book.enDesc || book.enShortDesc,
+            heDesc: book.heDesc || book.heShortDesc,
+            title: book.title,
+            heTitle: book.heTitle,
+            topicCat: book.categories[0],
+            heTopicCat: Sefaria.toc.filter(cat => cat.category === book.categories[0])[0].heCategory,
+            url: "/" + book.title,
+            analyticCat: "Book"
+        }
+    }
+    addTOCCategoryTopic(topic) {
+        const topicKeyArr = topic.key.slice();
+        const lastCat = topicKeyArr.pop(topicKeyArr - 1); //go up one level in order to get the bottom level's description
+        const relevantCats = topicKeyArr.length === 0 ? Sefaria.toc : Sefaria.tocItemsByCategories(topicKeyArr);
+        const relevantSubCat = relevantCats.filter(cat => "category" in cat && cat.category === lastCat)[0];
+        return {
+            analyticCat: "Category",
+            url: "/texts/" + topic.key.join("/"),
+            topicCat: "Texts",
+            heTopicCat: Sefaria.hebrewTerm("Texts"),
+            enDesc: relevantSubCat.enDesc,
+            heDesc: relevantSubCat.heDesc,
+            title: relevantSubCat.category,
+            heTitle: relevantSubCat.heCategory
+        }
+    }
+    async addGeneralTopic(topic) {
+        const d = await Sefaria.getTopic(topic.key, {annotated: false});
+        let searchTopic = {
+            analyticCat: "Topic",
+            title: d.primaryTitle["en"],
+            heTitle: d.primaryTitle["he"],
+            numSources: 0,
+            numSheets: 0,
+            url: "/topics/" + topic.key
+        }
+        const typeObj = Sefaria.topicTocCategory(topic.key);
+        if (!typeObj) {
+            searchTopic.topicCat = "Topics";
+            searchTopic.heTopicCat = Sefaria.hebrewTranslation("Topics");
+        } else {
+            searchTopic.topicCat = typeObj["en"];
+            searchTopic.heTopicCat = typeObj["he"];
+        }
+        if ("description" in d) {
+            searchTopic.enDesc = d.description["en"];
+            searchTopic.heDesc = d.description["he"];
+        }
+        if (d.tabs?.sources) {
+            searchTopic.numSources = d.tabs.sources.refs.length;
+        }
+        if (d.tabs?.sheets) {
+            searchTopic.numSheets = d.tabs.sheets.refs.length;
+        }
+        return searchTopic;
+    }
+    async addCollection(collection) {
+        const d = await Sefaria.getCollection(collection.key);
+        return {
+            analyticCat: "Collection",
+            title: d.name,
+            heTitle: d.name,
+            url: "/collections/" + collection.key,
+            topicCat: "Collections",
+            heTopicCat: Sefaria.hebrewTranslation("Collections"),
+            enDesc: d.description,
+            heDesc: d.description,
+            numSheets: d.sheets.length
+        }
+    }
+    async _executeTopicQuery() {
```

Comment:
Move all of the topic / ref / category getting functions to own class in this file. Mark them all as private except `_executeTopicQuery()`.